api/tests/unit/mappers/test_unit_mappers_dynamodb.py

```diff
@@ -0,0 +1,203 @@
+    )
+    map_environment_to_engine_mock.return_value = mock_engine_model
+    map_value_to_document_value_mock.return_value = "foobar"
+
+    # When
+    result = dynamodb.map_environment_to_environment_document(environment)
+
+    # Then
+    assert result == {"field": "foobar"}
+    map_environment_to_engine_mock.assert_called_once_with(environment)
+    map_value_to_document_value_mock.assert_called_once_with(mock_engine_model.field)
+
+
+def test_map_environment_api_key_to_environment_api_key_document__call_expected(
+    environment_api_key: "EnvironmentAPIKey",
+    mocker: MockerFixture,
+    mock_engine_model: BaseModel,
+) -> None:
+    # Given
+    map_environment_api_key_to_engine_mock = mocker.patch(
+        "mappers.dynamodb.map_environment_api_key_to_engine"
+    )
+    map_value_to_document_value_mock = mocker.patch(
+        "mappers.dynamodb._map_value_to_document_value"
+    )
+    map_environment_api_key_to_engine_mock.return_value = mock_engine_model
+    map_value_to_document_value_mock.return_value = "foobar"
+
+    # When
+    result = dynamodb.map_environment_api_key_to_environment_api_key_document(
+        environment_api_key,
+    )
+
+    # Then
+    assert result == {"field": "foobar"}
+    map_environment_api_key_to_engine_mock.assert_called_once_with(environment_api_key)
+    map_value_to_document_value_mock.assert_called_once_with(mock_engine_model.field)
+
+
+def test_map_engine_identity_to_identity_document__call_expected(
+    mocker: MockerFixture,
+    mock_engine_model: BaseModel,
+) -> None:
+    # Given
+    map_value_to_document_value_mock = mocker.patch(
+        "mappers.dynamodb._map_value_to_document_value"
+    )
+    map_value_to_document_value_mock.return_value = "foobar"
+
+    # When
+    result = dynamodb.map_engine_identity_to_identity_document(
+        mock_engine_model,
+    )
+
+    # Then
+    assert result == {"field": "foobar"}
+    map_value_to_document_value_mock.assert_called_once_with(mock_engine_model.field)
+
+
+def test_map_identity_to_identity_document__call_expected(
+    identity: "Identity",
+    mocker: MockerFixture,
+    mock_engine_model: BaseModel,
+) -> None:
+    # Given
+    document = {"field": "foobar"}
+    map_identity_to_engine_mock = mocker.patch(
+        "mappers.dynamodb.map_identity_to_engine"
+    )
+    map_engine_identity_to_identity_document_mock = mocker.patch(
+        "mappers.dynamodb.map_engine_identity_to_identity_document"
+    )
+    map_identity_to_engine_mock.return_value = mock_engine_model
+    map_engine_identity_to_identity_document_mock.return_value = document
+
+    # When
+    result = dynamodb.map_identity_to_identity_document(identity)
+
+    # Then
+    assert result == document
+    map_identity_to_engine_mock.assert_called_once_with(identity)
+    map_engine_identity_to_identity_document_mock.assert_called_once_with(
+        mock_engine_model,
+    )
+
+
+def test_map_value_to_document_value__return_expected() -> None:
```

Comment:
This test is a bit of a mystery to me. I'm not entirely sure what all the set up data is supposed to represent? There seems to be a mix of actual engine models (e.g. MV models) and test models. Maybe this is all valid, but it would probably benefit from some better comments if so.

Reply:
Fair enough. I'll address this comment along with [private method testing concerns](https://github.com/Flagsmith/flagsmith/pull/2290#discussion_r1242474066) and the [mocking data concern](https://github.com/Flagsmith/flagsmith/pull/2290#discussion_r1242468481) by rewriting these tests to use real inputs.

Reply:
Done.